### PR TITLE
Implemented add and remove functionality for todo-list

### PR DESCRIPTION
**Pull request by Dagic-zewdu**
**In this milestone, This functionality are added**

- Added  remove and add todo list functionality

 > Completed functionality has a stroke on their description

 > Completed button will remove a completed task

 > Reset button will remove all todos

- The create, update and delete from the array is done with a [separate js file](https://github.com/Dagic-zewdu/Todo-list/blob/todo-list/src/js/controller/action.js)

### CODE REVIEW

File: src/js/controller/action.js [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
 };
 
-export const removeCompleted = (list) => list.filter((todo) => !todo.completed);
\ No newline at end of file
+export const removeCompleted = (list) => {
+  const todos = ArrangeList(list.filter((todo) => !todo.completed));
+  AddtoStorage(todos);
+  RenderList(todos);
+  return todos;
+};
```
Comment (mikethreels) [inline src/js/controller/action.js]: - [ ] according to the requirements the user should have the possibility to remove 1 single list item independently, right now it is only possible to remove the list item if it is marked as completed by using the butter remove completed. 
- [ ] there is also no way for the user to edit the existing note, please create the possibility for them to do so.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Dagi, 

Good job on creating the function for adding a list item!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/controller/action.js [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
-    if (todo.index === id) { return { ...todo, completed: value }; }
+    if (todo.index.toString() === id) { return { ...todo, completed: value }; }
     return todo;
   });
   return todos;
 };
 
-export const removeCompleted = (list) => list.filter((todo) => !todo.completed);
\ No newline at end of file
+export const removeCompleted = (list) => {
```
Comment (elyor-doniyorov) [inline src/js/controller/action.js:18]: - [ ] This is great that you have implemented the editing functionality requested by the previous reviewer. However, deleting a single item clicking on a trash icon buton is still not addressed yet. Kindly implement that feature too. Take a look at [this video](https://www.youtube.com/watch?v=AcUd-_Yjjqg) to see the feature.

**Review by elyor-doniyorov — CHANGES_REQUESTED** [review]
Hi @Dagic-zewdu, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
## Hi @Dagic-zewdu, 

# Status: Approved ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**